compose.py

```diff
@@ -0,0 +1,248 @@
+#!/usr/bin/env python3
+# Copyright 2020-2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+import argparse
+import os
+import subprocess
+import sys
+import docker
+
+### Global variables
+TRITON_VERSION_MAP = {
+    '2.12.0dev': (
+        '21.07dev',  # triton container
+        '21.06'),  # upstream container
+}
+
+FLAGS = None
+
+#### helper functions
+def log(msg, force=False):
+    if force or not FLAGS.quiet:
+        try:
+            print(msg, file=sys.stderr)
+        except Exception:
+            print('<failed to log>', file=sys.stderr)
+
+
+def log_verbose(msg):
+    if FLAGS.verbose:
+        log(msg, force=True)
+
+
+def fail(msg):
+    print('error: {}'.format(msg), file=sys.stderr)
+    sys.exit(1)
+    
+
+def fail_if(p, msg):
+    if p:
+        fail(msg)
+        
+
+##### create base image for gpu
+def start_gpu_dockerfile(ddir, argmap, dockerfile_name):
+    # Set enviroment variables
+    df = '''
+#
+# Multistage build.
+#
+FROM nvcr.io/nvidia/tritonserver:{}-py3 as full
+FROM nvcr.io/nvidia/tritonserver:{}-py3-min
+ARG TRITON_VERSION={}
```

Comment:
Isn't this version information added by the common function shared with build.py? Why are you doing it again here?

Reply:
yep need to delete. Thanks for catching